src/peasy/PeasyCam.java

```diff
@@ -581,21 +579,53 @@ public void setRotations(final double pitch, final double yaw, final double roll
 		return new float[] { 0, 0, 0 };
 	}
 
+	
+	
+	
+	private boolean pushed_lights = false;
+
 	/**
-	 * Thanks to A.W. Martin for the code to do HUD
+	 * 
+	 * begin screen-aligned 2D-drawing.
+	 * <pre>
+	 * beginHUD()
+	 *   disabled depth test
+	 *   disabled lights
+	 *   ortho
+	 * endHUD()
+	 * </pre>
+	 * 
 	 */
 	public void beginHUD() {
-		g.pushMatrix();
 		g.hint(PConstants.DISABLE_DEPTH_TEST);
-		// Load the identity matrix.
+		g.pushMatrix();
 		g.resetMatrix();
-		// Apply the original Processing transformation matrix.
-		g.applyMatrix(originalMatrix);
+		if(g.isGL()){
+			PGraphicsOpenGL pgl = (PGraphicsOpenGL)g;
+			pgl.pushProjection();
+			pushed_lights = pgl.lights;
+			pgl.lights = false;
+		}
+		if(g.is3D()){
+			g.ortho(0, g.width, -g.height, 0, -Float.MAX_VALUE, +Float.MAX_VALUE);
```

Comment:
Does this not need to be saved/restored?

Reply:
Still need an answer to this question.

Reply:
I'n not exactly sure what you are referring too. All states are saved restored.

Reply:
Checked a couple of times now and cant see whats missing:

https://github.com/diwi/peasycam/blob/6c069f210ee7709cb0dbdd91b240f6ae60792700/src/peasy/PeasyCam.java#L599-L628